phase-4-explainer.md

```diff
@@ -0,0 +1,175 @@
+    about the semantic properties of an element or a page.
+  * build a reliable browser-based assistive technology -
+    for example, a browser extension which uses the accessibility tree
+    to implement a screen reader, screen magnifier, or other assistive functionality;
+    or an in-page tool.
+  * detect whether an accessibility property
+    has been successfully applied
+    (via ARIA or otherwise)
+    to an element -
+    for example, to detect whether a browser has implemented a particular version of ARIA.
+  * do any kind of console-based debugging/checking of accessibility tree issues.
+  * react to accessibility tree state,
+    for example, detecting the exposed role of an element
+    and modifying the accessible help text to suit.
+
+#### Why is accessing the computed properties being addressed last?
+
+**Consistency**
+Currently, the accessibility tree is not standardized between browsers:
+Each implements accessibility tree computation slightly differently.
+In order for this API to be useful,
+it needs to work consistently across browsers,
+so that developers don't need to write special case code for each.
+
+We want to take the appropriate time to ensure we can agree
+on the details for how the tree should be computed
+and represented.
+
+**Performance**
+Computing the value of many accessible properties requires layout.
+Allowing web authors to query the computed value of an accessible property
+synchronously via a simple property access
+would introduce confusing performance bottlenecks.
+
+We will likely want to create an asynchronous mechanism for this reason,
+meaning that it will not be part of the `accessibleNode` interface.
+
+**User experience**
+Compared to the previous three phases,
+accessing the computed accessibility tree will have the least direct impact on users.
+In the spirit of the [Priority of Constituencies](https://www.w3.org/TR/html-design-principles/#priority-of-constituencies),
+it makes sense to tackle this work last.
+
+
+#### ComputedAccessibleNode
+It is currently possible to check accessible attributes for a DOM element by using the `getAttribute` method:
+```js
+element.hasAttribute("aria-labelledby");
+element.getAttribute("aria-labelledby");
+```
+However, determining which accessible properties apply to an element can be cumbersome, as there can be multiple
+Accessing the computed accessible properties of a DOM element will be designed to mimic the behaviour of `getComputedStyle()`, but implemented as an asynchronous Promise-based API. A `ComputedAccessibleNode` can be requested for a specific element from the window:
+
+```html
+<button id="composeBtn">Compose</button>
+```
+```js
+var composeBtn = document.getElementById('composeBtn');
+window.getComputedAccessibleNode(element).then(
+  function(computedAccessibleNode) {
+    console.log(computedAccessibleNode.role);  // resolved!
+  },
+  function(error) {
+    console.log(error);  // rejected, but why?
```

Comment:
One possibility: what if `element` is null, or, like, some other kind of object? Would that reject?

Reply:
If it is null, it fails to call the method.

![screenshot from 2018-02-19 09-54-52](https://user-images.githubusercontent.com/36247690/36358006-01e54f06-155b-11e8-9bf4-371485e34b22.png)